bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/impl/ProductGraphqlServicImpl.java

```diff
@@ -0,0 +1,151 @@
+        return assets.parallelStream()
+                .filter(e -> !e.getDisabled() && e.getMediaType().equals("image"))
+                .map(this::mapAsset)
+                .sorted(Comparator.comparing(Asset::getPosition))
+                .collect(Collectors.toList());
+    }
+    @Override
+    public ProductInterface getProduct(GraphqlClient client, String sku  ) {
+        FilterTypeInput input = new FilterTypeInput().setEq(sku);
+        ProductFilterInput filter = new ProductFilterInput().setSku(input);
+        QueryQuery.ProductsArgumentsDefinition searchArgs = s -> s.filter(filter);
+        ProductsQueryDefinition queryArgs = q -> q.items(this.generateProductQuery());
+        String queryString = Operations.query(query -> query
+                .products(searchArgs, queryArgs)
+                .storeConfig(this.generateStoreConfigQuery())).toString();
+        GraphqlResponse<Query, Error> response = client.execute(new GraphqlRequest(queryString), Query.class, Error.class, QueryDeserializer.getGson());
+        Query rootQuery = response.getData();
+        List<ProductInterface> products = rootQuery.getProducts().getItems();
+        mediaBaseUrl = rootQuery.getStoreConfig().getSecureBaseMediaUrl();
+        if (products.size() > 0) {
+            return products.get(0);
+        }
+        return null;
+    }
+
+    public ProductPricesQueryDefinition generatePriceQuery() {
+        return q -> q
+                .regularPrice(rp -> rp
+                        .amount(a -> a
+                                .currency()
+                                .value()));
+    }
+
+    public SimpleProductQueryDefinition generateSimpleProductQuery() {
+        return q -> q
+                .id()
+                .sku()
+                .name()
+                .description(d -> d.html())
+                .image(i -> i.label().url())
+                .thumbnail(t -> t.label().url())
+                .urlKey()
+                .stockStatus()
+                .color()
+                .price(this.generatePriceQuery())
+                .mediaGalleryEntries(g -> g
+                        .disabled()
+                        .file()
+                        .label()
+                        .position()
+                        .mediaType());
+    }
+
+    public ProductInterfaceQueryDefinition generateProductQuery() {
+        // Custom attributes or attributes that are part of a non-standard
+        // attribute set have to be added to the query manually. This also
+        // requires the customer to use newly generated GraphQL classes.
+        return q -> q
+                .id()
+                .sku()
+                .name()
+                .image(i -> i.label().url())
+                .thumbnail(t -> t.label().url())
+                .urlKey()
+                .stockStatus()
+                .price(this.generatePriceQuery())
+                .categories(c -> c.urlPath())
+                .mediaGalleryEntries(g -> g
```

Comment:
I think we just need to query for the small or swatch image instead of loading the entire media gallery entries.